api/policies/v1/policyserver_webhook_test.go

```diff
@@ -92,8 +93,78 @@ func TestValidateMinAvailable(t *testing.T) {
 	require.ErrorContains(t, err, "minAvailable and maxUnavailable cannot be both set")
 }
 
+func TestValidateImagePullSecret(t *testing.T) {
+	tests := []struct {
+		name   string
+		secret *corev1.Secret
+		valid  bool
+	}{
+		{
+			"non existing secret",
+			nil,
+			false,
+		},
+		{
+			"secret of wrong type",
+			&corev1.Secret{
+				Type: "Opaque",
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test",
+				},
+			},
+			false,
+		},
+		{
+			"valid secret",
+			&corev1.Secret{
+				Type: "kubernetes.io/dockerconfigjson",
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test",
+					Namespace: "default",
+				},
+			},
+			true,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			k8sClient := fake.NewClientBuilder().Build()
+
+			if test.secret != nil {
+				err := k8sClient.Create(context.TODO(), test.secret)
+				if err != nil {
+					t.Errorf("failed to create secret: %s", err.Error())
+				}
+			}
+
+			policyServer := &PolicyServer{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "policy-server",
+					Namespace: "default",
+				},
+				Spec: PolicyServerSpec{
+					ImagePullSecret: "test",
+				},
+			}
+
+			policyServerValidator := policyServerValidator{
+				k8sClient:            k8sClient,
+				deploymentsNamespace: "default",
+			}
+			err := policyServerValidator.validate(context.Background(), policyServer)
```

Comment:
does it make sense to have this test here? IIUC this validate() should run our validateImagePullSecret() but I doubt that's happening.